src/installer/corehost/cli/json_parser.cpp

```diff
@@ -130,5 +135,41 @@ bool json_parser_t::parse_stream(pal::istream_t& stream,
     realloc_buffer(stream_size - current_pos);
     stream.read(m_json.data(), stream_size - current_pos);
 
-    return parse_json(context);
+    parse_json(m_json.data());
+    return validate_json(m_json.data(), m_json.size(), context);
+}
+
+bool json_parser_t::parse_file(const pal::string_t& path)
+{
+    // This code assumes that the caller has checked that the file `path` exists
+    // either within the bundle, or as a real file on disk.
+
+    if (bundle::info_t::is_single_file_bundle())
+    {
+        const bundle::location_t *location = nullptr;
+        const char* data = (const char*) bundle::info_t::config_t::map(path, location);
+
+        if (data != nullptr)
+        {
+#ifdef _WIN32 // _WIN32
+            // Windows doesn't use in-situ parsing. Therefore, pass the memory mapped data directly.
+            // The parser will then copy/convert the data appropriately.
+            parse_json(data);
+            bool result = validate_json(data, location->size, path);
+#else //  _WIN32
+            // On Linux, copy the input data, since m_document.ParseInsitu
+            // requires a mutable data input.
+            realloc_buffer(location->size);
+            memcpy(m_json.data(), data, location->size);
```

Comment:
I usually see the idiom of using the `swap()` method to initialize a `std::vector` from an array, so I was curious to see how Clang would generate the code. [It's identical](https://gcc.godbolt.org/z/9P7Qn_). I'm impressed. :)